### PR TITLE
Implement GitHub actions

### PR DESCRIPTION
**Pull request by jared-w**
This PR should definitely be squashed.

Further steps beyond this are:

- Once the `.github/release-drafter.yml` and `.github/pr-labeler.yml` files are in master, I can start testing those workflows in a follow up branch.
- The Dockerfile now needs some love and polish and the action that automatically builds the Dockerfile will help make sure it stays up to date
- Bikeshedding about `pr-labeler` and `release-drafter` in general. They were added as a low effort way to ease the creation of releases and I guessed at how people would use them.
- Improving and tweaking the scheduled and other hard-to-test workflows. I did my best to test these, some in a different repo, but it's difficult to cover every edge case without trying it out on the actual code.

And finally, removing the Makefile, travis, and appveyor configuration files.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -15,9 +15,9 @@ ARG DIFF="diff"
 ARG IGNORE_EXPECTED="--ignore-expected"
 ARG CABAL_BUILD_FLAGS="-j"
 ARG CABAL_INSTALL_FLAGS="${CABAL_BUILD_FLAGS}"
-RUN make tarball
+RUN ./cry build
```
Comment (atomb) [inline Dockerfile:18]: Thanks for doing this! I'm glad it works.

**Comment by atomb** [comment]
Oh, it might make sense to remove the Travis and AppVeyor configurations as part of this PR.

**Comment by brianhuffman** [comment]
@atomb @jared-w Since this was merged, I've received 8 notification emails (so far) about bogus build failures. They all have "view results" links like this one: https://github.com/GaloisInc/cryptol/actions/runs/105778058

All the results pages say "No jobs were run", and they have an "Annotations" box that says,
```
Error
Only actions in "GaloisInc" are allowed for this repository
```

**Comment by atomb** [comment]
I just noticed this, too, and don't entirely understand the output. However, there was just a conversation internally about this and I think Dan changed the account to allow outside actions. So this might just go away now.
